pkg/hostagent/pods.go

```diff
@@ -355,7 +355,7 @@ func (agent *HostAgent) syncEps() bool {
 				if err != nil {
 					opflexEpLogger(agent.log, ep).
 						Error("Error writing EP file: ", err)
-				} else if wrote || !ep.registered {
+				} else if wrote {
```

Comment:
Why was this check removed as part of this change?

Reply:
This was initially removed for testing creation of podif CRD in controller,  since ep-registry in nil for on prem case. @jayaramsatya Can you please comment on this? Now that we have the flag for on prem case turned on in ep-registry, should I revert this check?

Reply:
initially just to test the pod if creates I have asked to remove this. now that you have enabled the podif creations default you can revert this change

Reply:
Thanks @jayaramsatya